### PR TITLE
Print response from http call to keycloak server

### LINKED ISSUE #13557: Should print response from http call to keycloak server in  keycloak-authz-client-18.0.2.jar
### Description

Description：
When I try app-authz-rest-springboot example, if keycloak.credentials.secret is incorrect, accessing api endpoint will fail with exception message: 

`Caused by: org.keycloak.authorization.client.util.HttpResponseException: Unexpected response from server: 401 / Unauthorized`

but this is not enough to locate the root cause, I need to debug the code and check the server response.

so, I think it's a good idea to add http response info to execute method within HttpMethod class

`

> 
> if (statusCode < 200 || statusCode >= 300) {
>           throw new HttpResponseException("Unexpected response from server: " + statusCode + " / " + statusLine.getReasonPhrase(), statusCode, statusLine.getReasonPhrase(), bytes);
> }
> 

`

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by kyao233**
Closes #13557

### CODE REVIEW

**Comment by kyao233** [comment]
@DGuhr Can you help to check this one? thanks.

**Review by stianst — CHANGES_REQUESTED** [review]
Including the response in the stack trace is not ideal for a few reasons. It can have a performance impact, but more importantly there's a risk of leaking sensitive information in logs.

I would say we should have a separate log category for response logging, at debug or even tracing level that can be enabled explicitly.

File: authz/client/src/main/java/org/keycloak/authorization/client/util/HttpMethod.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public R execute(HttpResponseProcessor<R> responseProcessor) {
             int statusCode = statusLine.getStatusCode();
 
             if (statusCode < 200 || statusCode >= 300) {
-                throw new HttpResponseException("Unexpected response from server: " + statusCode + " / " + statusLine.getReasonPhrase(), statusCode, statusLine.getReasonPhrase(), bytes);
+                throw new HttpResponseException("Unexpected response from server: " + statusCode + " / " + statusLine.getReasonPhrase()  + " Response : " + new String(bytes), statusCode, statusLine.getReasonPhrase(), bytes);
```
Comment (pedroigor) [inline authz/client/src/main/java/org/keycloak/authorization/client/util/HttpMethod.java:95]: For debugging purposes, we can use Java Logging and log the response accordingly when level `FINE` is set.

**Comment by pedroigor** [comment]
@kyao233 Thanks for the contribution. I'm closing it for now but if you have time for applying the suggested changes, feel free to re-open it.

**Pull request closed without merging**